engine/runtime-compiler/src/test/java/org/enso/compiler/test/pass/PassPersistanceTest.java

```diff
@@ -33,6 +39,42 @@ public void cachePreferences() throws Exception {
     assertNotSame("But not ==", pref, out);
   }
 
+  @Test
+  public void graphScopePersistance() throws Exception {
+    var b = GraphBuilder.create();
+    var x = b.newDef("x", null, Option.empty());
+    var b2 = b.addChild();
+    var y = b2.newDef("y", null, Option.empty());
+    var ux = b2.newUse("x", null, Option.empty(), true);
+    var uy = b2.newUse("y", null, Option.empty(), true);
+
+    assertGraphScopePersistance("Before serialization", b.toGraph(), y.id());
+
+    var g = serde(Graph.class, b.toGraph(), -1);
```

Comment:
- This test is in the core of the fix
- the "shape" of `Graph` must be the same
- **before** and **after** serialization
- e.g. the same `assertGraphXyz` tests have to pass
- prior to this PR that wasn't the case